src/Voron/Impl/EncryptionBuffersPool.cs

```diff
@@ -102,7 +109,9 @@ public EncryptionBuffersPool(bool registerLowMemory = true, bool registerCleanup
                 thread.Allocations += size;
 
                 Debug.Assert(size == allocation.Size, $"size ({size}) == allocation.Size ({allocation.Size})");
-
+#if VALIDATE
+                pager?.UnprotectPageRange(allocation.Ptr, (ulong)size);
```

Comment:
At this level we don't know if the encryption buffer is gonna be used for only for reading or also for writing. This means that we won't prevent from modifying the encryption buffer even if it was requested for read only purposes.

I see that this is called from `AcquirePagePointerForNewPage()` and `AcquirePagePointer()` (via `GetBufferAndAddToTxState()`). Maybe we should `UnprotectPageRange()` only when requesting for a _new_ page?

Reply:
From the point of view of the buffers here, they are always for writing.
Since we copy the decrypted data to them.

Reply:
Maybe we could try to apply protect / unprotect at the level of CryptoPager then? I think we know there if page is for reading (`AcquirePagePointer()`) or for write (`AcquirePagePointerForNewPage()`) - I haven't checked that in detail.

My concern is that current implementation of VALIDATE for encryption guards only against modifying the buffers that are in the pool only.

Reply:
I don't think it matters that much. Up until this fix, we didn't have it at all.
And encryption buffers are already protected in this mode from buffer overruns.

Given that this is just an extension for the current behavior, I would say skipping it fine.

Reply:
Fine by me. But we need to remember that VALIDATE for encrypted and non encrypted environments isn't the same level of protection.